lib/apiservers/engine/backends/system.go

```diff
@@ -240,7 +241,8 @@ func (s *System) UnsubscribeFromEvents(chan interface{}) {
 }
 
 func (s *System) AuthenticateToRegistry(ctx context.Context, authConfig *types.AuthConfig) (string, string, error) {
-	return "", "", fmt.Errorf("%s does not implement System.AuthenticateToRegistry", ProductName())
+	defer trace.End(trace.Begin(""))
+	return RegistryService.Auth(authConfig, dockerversion.DockerUserAgent(ctx))
```

Comment:
We don't want or need CIDR format at this point - we're specifying registries via fqdn or IP.
Mirrors are irrelevant from a login perspective and likely a security issue given you then need to trust all of the mirrors.

Longer term we're going to want the following services in a VCH:
a. port layer API exposed to provide basic container ops
b. registry service that can, given an image tag, obtain the image and pass it to the port layer
c. personality services on top of a & b

We pretty much have to go with whatever is expedient at this point given the bundle timelines, but we do need to remember that this isn't supposed to be a direct re-implementation of the docker daemon function.